File: wasm-size/pi_approx/pi_approx.zig
const std = @import("std");

// Leibniz formula: π/4 = 1 - 1/3 + 1/5 - 1/7 + ...
pub fn main() void {
    var pi: f64 = 0.0;
    var sign: f64 = 1.0;
    var i: i32 = 0;
    while (i < 1000000) : (i += 1) {
        pi += sign / (2.0 * @as(f64, @floatFromInt(i)) + 1.0);
        sign = -sign;
    }
    pi *= 4.0;
    std.debug.print("pi = {d:.15}\n", .{pi});
}

File: wasm-size/pi_approx/pi_approx.rs
// Leibniz formula: π/4 = 1 - 1/3 + 1/5 - 1/7 + ...
fn main() {
    let mut pi = 0.0_f64;
    let mut sign = 1.0_f64;
    for i in 0..1_000_000 {
        pi += sign / (2.0 * i as f64 + 1.0);
        sign = -sign;
    }
    pi *= 4.0;
    println!("pi = {:.15}", pi);
}
